lib/ansible/modules/cloud/amazon/code_pipeline.py

```diff
@@ -0,0 +1,293 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: code_pipeline
+short_description: Create or delete AWS CodePipeline
+notes:
+    - for details of the parameters and returns see U(http://boto3.readthedocs.io/en/latest/reference/services/codepipeline.html)
+description:
+    - Create or delete a CodePipeline on AWS.
+version_added: "2.6"
+author:
+    - Stefan Horning (@stefanhorning) <horning@mediapeers.com>
+requirements: [ json, botocore, boto3 ]
+options:
+    name:
+        description:
+            - Name of the pipeline
+        required: true
+    role_arn:
+        description:
+            - ARN of the IAM role to use when executing the pipeline
+        required: true
+    artifact_store:
```

Comment:
It would be nice to document this (and stages) with suboptions. Here is an example: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_instance.py#L48-L63